cmd/snap-confine/snap-confine.c

```diff
@@ -509,16 +542,58 @@ int main(int argc, char **argv)
 	return 1;
 }
 
-static void enter_classic_execution_environment(void)
+static void enter_classic_execution_environment(const sc_invocation * inv)
 {
```

Comment:
Not a blocker, but if adding things, maybe add a comment that main.c will reassociate for /run/snapd/ns before calling this? The comment doesn't have to be long, just a quick pointer.